### DISCUSSION TITLE
Getting the player to face a direction

### DISCUSSION THREAD

**Discussion by MrGeoTech** [Minestom/Minestom#637] [Help]
So I have a fake player and it is going to a position and then sending the main hand animation however it is supposed to look at the target before it attacks so it looks good. However, when I run this code, it doesn't seem to work as the head will just tick to the position and then go back to facing the direction it was in after it had finished moving.
```
this.lookAt(target.getPos());
this.getController().sendArmAnimation(Hand.MAIN);
MinecraftServer.getSchedulerManager()
        .scheduleNextTick(() -> target.damage(this.damage));
```

**Comment by TheMode** [comment]
You can cancel the player movement and resume a few ticks after the attack. May need a recording to better understand the issue.

**Reply by MrGeoTech** [reply]
Here is the code that initializes the movement so that you can get a little better understanding of what's going on. It has a few methods that are not in minestom but I think they are pretty self explanatory.
```
this.lookAt(castle.getPos());
Pos attackingPos = castle.getAttackingPos(position).withYaw(position.yaw()).withPitch(position.pitch());
this.moveToExact(attackingPos);
while (position.distance(attackingPos) > .5) {
    try {
        Thread.sleep(10);
    } catch (InterruptedException e) {
        e.printStackTrace();
    }
}
this.getNavigator().setPathTo(null);
this.target = target;
```
And a recording of it happening

https://user-images.githubusercontent.com/73970344/152186130-ab06771b-9758-4973-ba29-e888e28f525f.mp4

**Reply by MrGeoTech** [reply]
As far as I can tell, this issue can not be fixed. There is something that turns the player and it can't be canceled. I've tried everything I can think of.
